File: c_cuda/convergence.c
#include "comum.h"
//---check convergence---

void convergence(int itc, double error, double residual_p, double residual_u, double residual_v){
    itc++;
    error = fmax(residual_p, residual_u);
    error = fmax(residual_v, error);
}   

File: OpenACC_Gpu/convergence.f90
!--- check convergence ---
SUBROUTINE convergence(itc, error, residual_p, residual_u, residual_v)
    use comum
    IMPLICIT NONE
    
    integer :: i, j, itc
    real(8) :: error, residual_p, residual_u, residual_v

    itc = itc+1
    error = MAX(residual_u, residual_v, residual_p)

    !IF (MOD(itc,1000).EQ.0) THEN
    !    WRITE(*,*) itc, ' ', residual_u, residual_v, residual_p
    !ENDIF

    !OPEN(unit=549,file='data/error.dat',status='unknown',position='appEND')
    !WRITE(549,*) itc, ' ', residual_u, residual_v, residual_p
    !close(549)
   
RETURN
END SUBROUTINE convergence
